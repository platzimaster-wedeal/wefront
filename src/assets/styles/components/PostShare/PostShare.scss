@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.post-share {
  position: relative;
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  padding: $space-size-normal 0;
  margin: $space-size-normal 0;
  border-top: 1px solid $color-black-30;
  border-bottom: 1px solid $color-black-30;

  &__content {
    padding: 0 $space-size-normal;

    & p {
      padding-bottom: $space-size-small;
      @include font-size($font-size-small);

      @include media(tablet) {
        @include font-size($font-size-normal);
      }
    }

    &--image {
      position: relative;
      width: 100%;
      height: auto;

      & img {
        width: 100%;
        height: auto;
        object-fit: cover;
      }
    }
  }
}
