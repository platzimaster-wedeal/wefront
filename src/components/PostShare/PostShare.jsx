import React, { useState } from "react";

// Styles
import "../../assets/styles/components/PostShare/PostShare.scss";

// Components
import PostHeader from "../PostHeader/PostHeader";
import PostInteractions from "../PostInteractions/PostInteractions";
import PostComments from "../PostComments/PostComments";

const PostShare = ({
 post,
 comments = false,
}) => {
 const [seeComments, setSeeComments] = useState(true);

 return (
  <article className="post-share">
   <PostHeader 
    name={post.name}
    avatar={post.avatar}
    profession={post.profession}
    share 
    />
   <div className="post-share__content">
    <p>
     {post && post.content
      ? post.content
      : "Optional description until I know what to put here, besides, a need to know how it looks :D"}
    </p>
    <PostInteractions />
   </div>
   <div className="post-share__comments">
    {comments && seeComments && comments.length > 0 && (
     <PostComments comments={comments} isPost />
    )}
   </div>
  </article>
 );
};

export default PostShare;
