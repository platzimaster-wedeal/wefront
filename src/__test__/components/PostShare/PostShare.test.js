import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostShare from "../../../components/PostShare/PostShare";

describe("Test for the component PostShare :D", () => {
 const PostShareMounted = mount(<PostShare />);
 const PostShareCreated = create(<PostShare />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostShareCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostShareMounted.length).toEqual(1);
  });
  test("Should the content of the post be a string", () => {
   expect(
    typeof PostShareMounted.find(".post-share__content p").text()
   ).toEqual("string");
  });
 });
});
