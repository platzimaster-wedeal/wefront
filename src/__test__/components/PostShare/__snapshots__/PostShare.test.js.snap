// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostShare :D Visual tests for the component should the comopnent have the same tree 1`] = `
<article
  className="post-share"
>
  <section
    className="post-header"
  >
    <figure
      className="post-header__picture"
    >
      <img
        className="image image__isOnline undefined"
        src={Object {}}
      />
    </figure>
    <div
      className="post-header__user-info"
    >
      <div
        className="post-header__user-info--main"
      >
        <h3>
          Alan Walks
        </h3>
        <span>
          Has shared!
        </span>
      </div>
      <strong
        className="post-header__user-info--profession"
      />
    </div>
    <small
      className="post-header__time"
    >
      1hr
    </small>
  </section>
  <div
    className="post-share__content"
  >
    <p>
      Optional description until I know what to put here, besides, a need to know how it looks :D
    </p>
    <div
      className="interactionsPost__container"
    >
      <button
        className="interactionsPost__favs"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="interactionsPost__icon--inactive"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
          />
        </svg>
         Favs
      </button>
      <button
        className="interactionsPost__shares "
        onClick={[Function]}
        type="button"
      >
        <svg
          className="interactionsPost__icon--inactive"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
          />
        </svg>
         Shares
      </button>
      <button
        className="interactionsPost__comments"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="interactionsPost__icon--inactive"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h14l4 4V4c0-1.1-.9-2-2-2zm-2 12H6v-2h12v2zm0-3H6V9h12v2zm0-3H6V6h12v2z"
          />
        </svg>
         Comments
      </button>
      <button />
    </div>
  </div>
  <div
    className="post-share__comments"
  >
    <section
      className="post-comments"
    >
      <strong
        className="post-comments__action"
        onClick={[Function]}
      >
        See all comments
      </strong>
      <div
        className="post-comments__messages"
      >
        <article
          className="message undefined"
        >
          <figure
            className="message__photo"
          >
            <img
              className="image image__isOffline undefined"
              src={Object {}}
            />
          </figure>
          <div
            className="message__content undefined"
          >
            <div
              className="message__content--msg"
            >
              <strong>
                Tupac Shakur
              </strong>
              <p>
                Mesg
              </p>
            </div>
            <div
              className="message__content--actions"
            >
              <div
                className="message-actions"
              >
                <div
                  className="message-actions__fav"
                >
                  <div
                    className="graphic-data undefined"
                  >
                    <svg
                      fill="currentColor"
                      height="24"
                      size="24"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                      />
                    </svg>
                    <strong>
                      0 favs
                    </strong>
                  </div>
                </div>
                <div
                  className="message-actions__reply"
                >
                  <div
                    className="graphic-data undefined"
                  >
                    <svg
                      fill="currentColor"
                      height="24"
                      size="24"
                      stroke="currentColor"
                      strokeWidth="0"
                      style={
                        Object {
                          "color": undefined,
                        }
                      }
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
                      />
                    </svg>
                    <strong>
                      Reply
                    </strong>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </article>
      </div>
      <div
        className="post-comments__user-comment"
      >
        <img
          className="image image__isOffline undefined"
          src={Object {}}
        />
        <input
          placeholder="What are you thinking?"
          type="text"
        />
      </div>
    </section>
  </div>
</article>
`;
